app/assets/javascripts/components/sidebar/posts_main_component.js.coffee

```diff
@@ -0,0 +1,44 @@
+# @state.posts is an array of json objects describing posts
+window.PostsMainComponent = React.createClass
+	getInitialState: ->
+		{posts_json: []}
+
+	componentWillMount: ->
+		@loadPostsFromServer()
+	
+	componentDidMount:->
+		@pollServer()	
+
+	loadPostsFromServer: ->
+		console.log "loading posts from server"
+		url = 'posts' + (if @props.postsAreRelevant then '/relevant' else '')
+		return $.ajax
+			url: url+ '.json'
```

Comment:
shouldn't need the `+ '.json'` since you specify `dataType` but I might be incorrect about that.